### ISSUE TITLE
[Question] Auto screenshots run after test.afterEach, silently fails if page.close() is used

### ISSUE THREAD

**Issue by mrienstra** [microsoft/playwright#9247]
I was testing out various debugging options, and noticed that `testOptions.screenshot` (for [automatic screenshots](https://playwright.dev/docs/test-configuration#automatic-screenshots)) was not working (no `test-results` folder created), though `testOptions.trace` and `testOptions.video` were working fine.

Dug into it a little and realized the issue was that my test files had:
```ts
test.afterEach(async ({ page }) => {
  await page.close();
});
```
... resulting in `context.pages() === []` here: https://github.com/microsoft/playwright/blob/241411a/src/test/index.ts#L243

Now that I understand what's happening, no biggie, but I have two questions:

1. Would it make sense to offer a clue (console output) in this scenario (when `screenshot === 'on' || screenshot === 'only-on-failure'` but `context.pages() === []`), or is this too much of an edge case?
2. Almost seems like it would be more useful to capture the screenshot before `test.afterEach` runs. I don't have a particular use case in mind, it just feels more correct.

**Comment by udit0802** [comment]
I have also faced same issue, so instead of page.close, I am using context.close in aftereach method, as that seems correct as well, as each test should get a new context anyways, can you try that way.

**Comment by mrienstra** [comment]
@udit0802, interesting, that works, though I'm not sure why, I'll look into it tomorrow!

Specifically, switching from `page.close()` to [`page.context()`](https://playwright.dev/docs/api/class-page#page-context)`.close()`:
```ts
test.afterEach(async ({ page }) => {
  await page.close();
});
```
To:
```ts
test.afterEach(async ({ page }) => {
  await page.context().close();
});
```


I believe the hierarchy is `browser` --> `context` --> `page`, and [docs for `browserContext.close()`](https://playwright.dev/docs/api/class-browsercontext/#browser-context-close) say:
>Closes the browser context. All the pages that belong to the browser context will be closed.

... Which is why I'm surprised that helped. Thanks!

**Comment by udit0802** [comment]
> @udit0802, interesting, that works, though I'm not sure why, I'll look into it tomorrow!
> 
> Specifically, switching from `page.close()` to [`page.context()`](https://playwright.dev/docs/api/class-page#page-context)`.close()`:
> 
> ```ts
> test.afterEach(async ({ page }) => {
>   await page.close();
> });
> ```
> 
> To:
> 
> ```ts
> test.afterEach(async ({ page }) => {
>   await page.context().close();
> });
> ```
> 
> I believe the hierarchy is `browser` --> `context` --> `page`, and [docs for `browserContext.close()`](https://playwright.dev/docs/api/class-browsercontext/#browser-context-close) say:
> 
> > Closes the browser context. All the pages that belong to the browser context will be closed.
> 
> ... Which is why I'm surprised that helped. Thanks!

Yes I also tried the above solution based on then intuition from the statement pointed out by you.
```
Closes the browser context. All the pages that belong to the browser context will be closed.
```
Glad that helps.

**Comment by aslushnikov** [comment]
@udit0802 @mrienstra why do you want to close page / context? these will be automatically closed - you don't need to do this.

**Comment by udit0802** [comment]
> @udit0802 @mrienstra why do you want to close page / context? these will be automatically closed - you don't need to do this.

Just to highlight , I am closing the context because if we don't then till the time all the tests are not executed the window remains open for the previous test execution, putting extra load(resource intensive) on the testing machine.
If I use context.close(), then window is closed after the test run execution which is the desired behaviour.
This behaviour I have observed on playwright version 1.15 as well.

**Comment by dgozman** [comment]
@udit0802 Could you please share your test? Do you call `newContext()` in your tests? Any context that was created for you by the test runner should be closed immediately after the test finishes, so we must have a bug unless you create context yourself.

**Comment by udit0802** [comment]
Yes i am creating context by myself using the browser provided by the test runner, as I believe every test should get different context for the execution.

**Comment by dgozman** [comment]
@udit0802 Default `page` and `context` are already isolated for each test, so you can just use them and simplify your test files. [Learn more here](https://playwright.dev/docs/intro#using-test-fixtures).
